klient/decoder/main_test.go

```diff
@@ -14,35 +14,39 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package decoder
+package decoder_test
```

Comment:
@harshanarayana I am coming back here so it doesn't fall the radar. `decoder_test`  means our dependency graph has an issue. Let's open a PR (if you haven't done so) for package re-org to avoid this situation.

Reply:
@vladimirvivien We will move that work of refactoring into a different PR. That might break some existing imports and others for the existing users. So better to track that as a different item?

Reply:
Agreed.